Lib/glyphsLib/builder/glyph.py

```diff
@@ -243,13 +246,14 @@ def to_ufo_glyph_background(self, glyph, layer):
     self.to_ufo_glyph_anchors(new_glyph, background.anchors)
     self.to_ufo_guidelines(new_glyph, background)
 
+def _to_glyphs_color(color):
+    # type: (defcon.objects.color.Color) -> Union[int, List[int]]
 
-def _to_glyphs_color(self, color):
-    # color is a defcon Color
-    # Try to find a matching Glyphs color
+    # If the color matches one of Glyphs's predefined colors, return that
+    # index.
     for index, glyphs_color in enumerate(GLYPHS_COLORS):
         if str(color) == glyphs_color:
             return index
 
-    # Otherwise, make a Glyphs-formatted color list
-    return [round(component * 255) for component in list(color)]
+    # Otherwise, make a Glyphs-formatted color list: [u8, u8, u8, 1].
+    return [round(component * 255) for component in tuple(color)[:4]]
```

Comment:
again, the UFO comma-separated color string always contains 4 items. if it doesn't it's invalid. I think defcon or ufoLib would choke on such an incomplete color string, or one that contains more than 4 items.
Do an assert if you want to be cautious, but remove that slice.